pkg/snapstore/s3_snapstore_test.go

```diff
@@ -46,11 +54,80 @@ func (m *mockS3Client) PutObject(in *s3.PutObjectInput) (*s3.PutObjectOutput, er
 	in.Body.Seek(0, io.SeekStart)
 	temp := make([]byte, off)
 	in.Body.Read(temp)
-	m.objects[*in.Key] = temp
+	m.objects[*in.Key] = &temp
 	out := s3.PutObjectOutput{}
 	return &out, nil
 }
 
+func (m *mockS3Client) CreateMultipartUploadWithContext(ctx aws.Context, in *s3.CreateMultipartUploadInput, opts ...request.Option) (*s3.CreateMultipartUploadOutput, error) {
+	uploadID := time.Now().String()
+	var parts [][]byte
+	m.multiPartUploads[uploadID] = &parts
+	out := &s3.CreateMultipartUploadOutput{
+		Bucket:   in.Bucket,
+		UploadId: &uploadID,
+	}
+	return out, nil
+}
+
+func (m *mockS3Client) UploadPartWithContext(ctx aws.Context, in *s3.UploadPartInput, opts ...request.Option) (*s3.UploadPartOutput, error) {
+	if m.multiPartUploads[*in.UploadId] == nil {
+		return nil, fmt.Errorf("multipart upload not initiated")
+	}
+	if *in.PartNumber < 0 {
+		return nil, fmt.Errorf("part number should be positive integer")
+	}
+	if *in.PartNumber > int64(len(*m.multiPartUploads[*in.UploadId])) {
```

Comment:
It can be more than one. Since all go routine for upload chunk run in parallel. So for 5 part chunk, it might happen that call for chunk with partNumber 5 may come immediately after partNumaber 1.  But at the end of all routines there won't be any gaps, Cause maxSize of array will be equal to noOfChunks, since partNumber will be sequential ranging from 1 to noOfChunks without any gaps.